lib/manageiq/api/client/resource.rb

```diff
@@ -0,0 +1,40 @@
+require_relative "mixins/action_mixin"
+
+module ManageIQ
+  module API
+    module Client
+      class Resource
+        include ActionMixin
+
+        attr_accessor :collection
+
+        delegate :server, :to => :@collection
+
+        def initialize(collection)
+          @collection = collection
+        end
+
+        def self.subclass(name)
+          klass_name = name.classify
+
+          if ManageIQ::API::Client::Resource.const_defined?(klass_name)
+            ManageIQ::API::Client::Resource.const_get(klass_name)
+          else
+            klass = Class.new(ManageIQ::API::Client::Resource) do
```

Comment:
Much like Collection...The Resource itself can have the properties, and then none of this is necessary.